craft_application/application.py

```diff
@@ -701,14 +701,6 @@ def _get_project_vars(self, yaml_data: dict[str, Any]) -> dict[str, str]:
             pvars[var] = str(yaml_data.get(var, ""))
         return pvars
 
-    def _set_global_environment(self, info: craft_parts.ProjectInfo) -> None:
```

Comment:
@lengau, how aggressive would you like to be with this?

I'm leaning towards removing it completely or raising an error.  A deprecation warning doesn't seem sufficient since it's not obvious to an application that overrides that it will never be used.

Reply:
@lengau and I discussed offline and agreed to raise a deprecation warning.